### PR TITLE
Add beta div plugin

### PR DESCRIPTION
**Pull request by asizemore**
The goal of this PR is to create a functional beta diversity app.

The app has a [sister plugin](https://github.com/VEuPathDB/EdaDataService/blob/master/src/main/java/org/veupathdb/service/eda/ds/plugin/betadiv/BetaDivScatterplotPlugin.java) in the data service that we can learn from. Importantly, the ds plugin should be removed after this compute service plugin has been completed.

The beta div plugin requires the following parameters
- collection variable: a category-type variable. For example, "Relative abundance of class data". The data passed into the beta div computation (R functions) is the set of variables that are children of this collection variable.
- beta div distance method: one of "jsd", "jaccard", "bray" that determines what type of dissimilarity to calculate (passed to R functions).

# To Dos
- [x] Create plugin skeleton
- [x] Have beta div function run in RServe with fake data
- [x] Use the input method in beta div R functions
- [x] Input real data (swap to `useRConnectionWithRemoteFiles`)
- [x] get/use the Sample Id column name from collection variable and metadata (currently using `computeEntityIdColName`)
- [x] write tabular output
- [x] write metadata output
- [x] testing

# Notes
1. I need to go back to microbiomeComputations and rework the functions a bit so that getting the metadata and tabular data from the R functions are easier (also noted in the beta div plugin comments)

# Example beta div request 
(`http://localhost:8080/computes/betadiv`)
```
{
    "studyId":"DailyBaby-1",
    "filters":[],
    "config":{
        "outputEntityId":"EUPATH_0000813",
        "collectionVariable":{
            "entityId":"EUPATH_0000813",
            "variableId":"EUPATH_0009252"
        },
        "betaDivDistanceMethod": "bray"
    }
}
```

### CODE REVIEW

**Comment by d-callan** [comment]
havent tested this yet
need to confirm data output format
need to figure out relationship between metadata and stats endpoints, how things like percent variance will eventually get to the client etc

**Comment by d-callan** [comment]
my current thinking is that the metadata endpoint will provide everything the client currently needs for ComputatedVariableMetadata, stats endpoint will be everything else.

File: src/main/java/org/veupathdb/service/eda/compute/plugins/betadiv/BetaDivPlugin.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  private static final String INPUT_DATA = "beta_div_input";
+
+  public BetaDivPlugin(@NotNull PluginContext<BetaDivPluginRequest, BetaDivPluginConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable().getEntityId())
```
Comment (ryanrdoherty) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/betadiv/BetaDivPlugin.java:30]: Is there a place in the compute service framework to validate the plugin request?  Thinking we should be checking if the input var is a collection var and maybe its type (number, integer, date?) if that matters.

**Review by ryanrdoherty — APPROVED** [review]

File: src/main/java/org/veupathdb/service/eda/compute/plugins/betadiv/BetaDivPlugin.java (hunk 1 above)
Comment (d-callan) [reply src/main/java/org/veupathdb/service/eda/compute/plugins/betadiv/BetaDivPlugin.java:30]: Nothing I saw yet, but I didn't look very hard tbh.

**Comment by d-callan** [comment]
this depends on veupathdb/microbiomecomputations#25

File: src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+
+      connection.voidEval("alphaDivDT <- alphaDiv(" + INPUT_DATA + ", " + 
+                                                    util.singleQuote(computeEntityIdColName) + ", " + 
+                                                    util.singleQuote(method) + ")");
+      String dataCmd = "print(alphaDivDT)";
+      String metaCmd = "getMetadata(alphaDivDT)";
```
Comment (bobular) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java:57]: I'm not sure if the alpha div R package exports a getMetaData?
[beta div does](https://github.com/VEuPathDB/microbiomeComputations/blob/574ebd79e6bea6ff25f63342215319c5bd0ce55c/R/app_betaDiv.R#L164)

**Review by bobular — COMMENTED** [review]
Just that one comment about `getMetadata()` really. I've had a look through both PRs and have a rough idea what's going on!

File: src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java (hunk 2 above)
Comment (d-callan) [reply src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java:57]: getMetadata is currently written as a generic fxn as opposed to a method specific to a class, the one defined in the beta div file will work for any of the computes currently. This is something I plan to refactor and do properly soon but don't want to hold up a release of beta div for.

Comment (bobular) [reply src/main/java/org/veupathdb/service/eda/compute/plugins/alphadiv/AlphaDivPlugin.java:57]: Ah, I see - cool!

**Comment by d-callan** [comment]
initial testing seems to have three working plugins :tada: , beta div currently in slow queue and the other two in the fast one

**Pull request merged**